spec/integration/bisect_spec.rb

```diff
@@ -0,0 +1,158 @@
+RSpec::Support.require_rspec_core "formatters/bisect_progress_formatter"
+
+module RSpec::Core
+  RSpec.describe "Bisect", :slow, :simulate_shell_allowing_unquoted_ids do
+    include FormatterSupport
+
+    before do
+      skip "These specs do not consistently pass or fail on AppVeyor on Ruby 2.1+"
+    end if ENV['APPVEYOR'] && RUBY_VERSION.to_f > 2.0
+
+    def bisect(cli_args, expected_status=nil)
+      RSpec.configuration.output_stream = formatter_output
+      parser = Parser.new(cli_args + ["--bisect"])
+      expect(parser).to receive(:exit).with(expected_status) if expected_status
+
+      expect {
+        parser.parse
+      }.to avoid_outputting.to_stdout_from_any_process.and avoid_outputting.to_stderr_from_any_process
+
+      normalize_durations(formatter_output.string)
+    end
+
+    it 'finds the minimum rerun command and exits' do
+      output = bisect(%w[spec/rspec/core/resources/order_dependent_specs.rb --order defined], 0)
+
+      expect(output).to eq(<<-EOS.gsub(/^\s+\|/, ''))
+        |Bisect started using options: "spec/rspec/core/resources/order_dependent_specs.rb --order defined"
+        |Running suite to find failures... (n.nnnn seconds)
+        |Starting bisect with 1 failing example and 21 non-failing examples.
+        |
+        |Round 1: searching for 11 non-failing examples (of 21) to ignore: .. (n.nnnn seconds)
+        |Round 2: searching for 6 non-failing examples (of 11) to ignore: . (n.nnnn seconds)
+        |Round 3: searching for 3 non-failing examples (of 5) to ignore: . (n.nnnn seconds)
+        |Round 4: searching for 1 non-failing example (of 2) to ignore: . (n.nnnn seconds)
+        |Round 5: searching for 1 non-failing example (of 1) to ignore: . (n.nnnn seconds)
+        |Bisect complete! Reduced necessary non-failing examples from 21 to 1 in n.nnnn seconds.
+        |
+        |The minimal reproduction command is:
+        |  rspec ./spec/rspec/core/resources/order_dependent_specs.rb[11:1,22:1] --order defined
+      EOS
+    end
+
+    it 'supports a verbose mode via `DEBUG_RSPEC_BISECT` so we can get detailed log output from users when they report bugs' do
```

Comment:
We could make this a scenario but I think I'd prefer not... (unless we add a huge caveat about it) as this is for debug purposes does it even need a spec?